sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -598,6 +598,11 @@ trait IntervalDivide {
       }
     }
   }
+
+  def divideByZeroCheckCodegen(expr: Expression, value: String): String = expr.dataType match {
```

Comment:
nit: Expression is too general as the first parameter, you could pass `DataType`. Up to you.